mlflow/gluon.py

```diff
@@ -0,0 +1,184 @@
+    """
+    Save a Gluon model to a path on the local file system.
+
+    :param gluon_model: Gluon model to be saved. Must be already hybridized.
+    :param path: Local path where the model is to be saved.
+    :param mlflow_model: MLflow model config this flavor is being added to.
+
+    >>> from mxnet.gluon import Trainer
+    >>> from mxnet.gluon.contrib import estimator
+    >>> from mxnet.gluon.loss import SoftmaxCrossEntropyLoss
+    >>> from mxnet.gluon.nn import HybridSequential
+    >>> from mxnet.metric import Accuracy
+    >>> import mlflow
+    >>> # Build, compile, and train your model
+    >>> gluon_model_path = ...
+    >>> net = HybridSequential()
+    >>> with net.name_scope():
+    >>> ...
+    >>> net.hybridize()
+    >>> net.collect_params().initialize()
+    >>> softmax_loss = SoftmaxCrossEntropyLoss()
+    >>> trainer = Trainer(net.collect_params())
+    >>> est = estimator.Estimator(net=net, loss=softmax_loss, metrics=Accuracy(), trainer=trainer)
+    >>> est.fit(train_data=train_data, epochs=100, val_data=validation_data)
+    ... # Save the model as an MLflow Model
+    >>> mlflow.gluon.save_model(net, gluon_model_path)
+    """
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise MlflowException("Path '{}' already exists".format(path))
+    data_subpath = "data"
+    data_path = os.path.join(path, data_subpath)
+    os.makedirs(data_path)
+    # The epoch argument of the export method does not play any role in selecting
+    # a specific epoch's paramaters, and is there only for display purposes.
+    gluon_model.export(os.path.join(data_path, _MODEL_SAVE_PATH))
+    with open(os.path.join(path, "architecture.txt"), "w") as fp:
+        fp.write(str(gluon_model))
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+@experimental
+def log_model(gluon_model: HybridBlock, artifact_path: str):
+    """
+    Log a Gluon model as an MLflow artifact for the current run.
+
+    :param gluon_model: Gluon model to be saved. Must be already hybridized.
+    :param artifact_path: Run-relative artifact path.
+
+    >>> from mxnet.gluon import Trainer
+    >>> from mxnet.gluon.contrib import estimator
+    >>> from mxnet.gluon.loss import SoftmaxCrossEntropyLoss
+    >>> from mxnet.gluon.nn import HybridSequential
+    >>> from mxnet.metric import Accuracy
+    >>> import mlflow
+    >>> # Build, compile, and train your model
+    >>> net = HybridSequential()
+    >>> with net.name_scope():
+    >>> ...
+    >>> net.hybridize()
+    >>> net.collect_params().initialize()
+    >>> softmax_loss = SoftmaxCrossEntropyLoss()
+    >>> trainer = Trainer(net.collect_params())
+    >>> est = estimator.Estimator(net=net, loss=softmax_loss, metrics=Accuracy(), trainer=trainer)
+    >>> est.fit(train_data=train_data, epochs=100, val_data=validation_data)
+    >>> # Log metrics and log the model
+    >>> with mlflow.start_run() as run:
```

Comment:
Probably this should wrap the fit call as well, right, to track at least the start/end times.

Reply:
I'm not sure I get this part?

Reply:
Just in the example, I mean it should probably be

```python
with mlflow.start_run() as run:
  est.fit(...)
  mlflow.log_mdoel
```

Reply:
Right, got it now. I'll just update the documentation.